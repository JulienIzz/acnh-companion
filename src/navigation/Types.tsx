export type RootStackParamList = {
  Home: undefined;
  Encyclopedie: undefined;
  Search: undefined;
  BugReporting: undefined;
};

export type Routes = keyof RootStackParamList;

declare global {
  namespace ReactNavigation {
    interface RootParamList extends RootStackParamList {}
  }
}
