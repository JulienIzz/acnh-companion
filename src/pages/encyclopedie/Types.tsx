export interface Fish {
  id: number;
  name: {
    'name-USen': string;
    'name-EUfr': string;
  };
  availability: {
    'month-northern': string;
    'month-southern': string;
    time: string;
    isAllDay: boolean;
    isAllYear: boolean;
    location: string;
    rarity: string;
    'month-array-northern': Array<number>;
    'month-array-southern': Array<number>;
  };
  shadow: string;
  price: number;
  'price-cj': number;
  icon_uri: string;
  image_uri: string;
}
