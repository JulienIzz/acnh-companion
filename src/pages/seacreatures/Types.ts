export type SeaCreature = {
  id: number;
  name: {
    'name-USen': string;
    'name-EUfr': string;
  };
  availability: {
    'month-northern': string;
    'month-southern': string;
    time: string;
    isAllDay: boolean;
    isAllYear: false;
    'month-array-northern': Array<number>;
    'month-array-southern': Array<number>;
    'time-array': Array<number>;
  };
  speed: string;
  shadow: string;
  price: number;
  image_uri: 'https://acnhapi.com/v1/images/sea/1';
  icon_uri: 'https://acnhapi.com/v1/icons/sea/1';
};
