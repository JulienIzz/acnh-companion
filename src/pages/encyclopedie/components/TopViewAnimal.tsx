import React from 'react';
import {View, Image} from 'react-native';
import {SimpleTypography} from '../../../theme/typography/Typography';
import {StyleSheet} from 'react-native';
import {Animal} from '../Types';

interface TopViewProps {
  animal: Animal;
}

export const TopViewAnimal = ({animal}: TopViewProps) => {
  return (
    <>
      <View style={styles.animalID}>
        <SimpleTypography text={'Nom : ' + animal.name['name-EUfr']} />
        <SimpleTypography text={'RaretÃ© : ' + animal.availability.rarity} />
        <SimpleTypography text={'Prix : ' + animal.price} />
      </View>
      <View style={styles.animalImageContainer}>
        <Image
          style={styles.animalImage}
          source={{
            uri: animal.icon_uri,
          }}
        />
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  animalCard: {
    width: '90%',
    alignSelf: 'center',
    backgroundColor: '#FAEDCD',
    borderColor: '#D4A373',
    borderRadius: 10,
    borderWidth: 3,
    margin: 5,
    padding: 8,
    flexDirection: 'row',
    alignItems: 'center',
  },
  animalID: {
    flex: 1,
    justifyContent: 'space-evenly',
  },
  animalImage: {
    width: '100%',
    height: '100%',
  },
  animalImageContainer: {
    width: 80,
    height: 80,
    margin: 5,
  },
});
