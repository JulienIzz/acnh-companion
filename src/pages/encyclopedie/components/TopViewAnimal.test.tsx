import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {TopViewAnimal} from './TopViewAnimal';
import {Fish} from '../Types';

const mockFishArray: Fish[] = [
  {
    id: 1,
    name: {
      'name-USen': 'PoissonPilote1',
      'name-EUfr': 'PoissonPilote1',
    },
    availability: {
      'month-northern': '4 - 9',
      'month-southern': '4 - 9',
      'month-array-northern': [1, 2],
      'month-array-southern': [1, 2],
      time: '9am - 4pm',
      isAllDay: false,
      isAllYear: false,
      location: 'River',
      rarity: 'Common',
    },
    'price-cj': 1200,
    icon_uri: '',
    image_uri: '',
    shadow: 'Small',
    price: 900,
  },
];

test('animal top view card with mock fish information', async () => {
  render(
    <TopViewAnimal
      name={mockFishArray[0].name['name-EUfr']}
      rarity={mockFishArray[0].availability.rarity}
      price={mockFishArray[0].price}
      icon={mockFishArray[0].icon_uri}
    />,
  );

  const nameOutput = screen.getByText('Nom : PoissonPilote1');
  const priceOutput = screen.getByText('Prix : 900');
  const rarityOutput = screen.getByText('Raret√© : Common');

  expect(nameOutput).toBeTruthy();
  expect(priceOutput).toBeTruthy();
  expect(rarityOutput).toBeTruthy();

  expect(screen.toJSON()).toMatchSnapshot();
});
