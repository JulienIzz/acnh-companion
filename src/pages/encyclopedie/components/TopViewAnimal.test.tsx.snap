// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`animal top view card with mock fish information 1`] = `
[
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "space-evenly",
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 14,
        }
      }
    >
      Nom : PoissonPilote1
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 14,
        }
      }
    >
      Raret√© : Common
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 14,
        }
      }
    >
      Prix : 900
    </Text>
  </View>,
  <View
    style={
      {
        "height": 80,
        "margin": 5,
        "width": 80,
      }
    }
  >
    <Image
      source={
        {
          "uri": "",
        }
      }
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>,
]
`;
